packages/react/src/components/ComboButton/index.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright IBM Corp. 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { ChevronDown } from '@carbon/icons-react';
+import { Button } from '../Button';
+import { IconButton } from '../IconButton';
+import { Menu } from '../Menu';
+
+import { useId } from '../../internal/useId';
+import { usePrefix } from '../../internal/usePrefix';
+import { useAttachedMenu } from '../../internal/useAttachedMenu';
+
+const spacing = 4; // top and bottom spacing between the button and the menu. in px
+
+function ComboButton({
+  children,
+  disabled,
+  kind = 'primary',
+  label,
+  onClick,
+  size = 'md',
+}) {
+  const id = useId('combobutton');
+  const prefix = usePrefix();
+
+  const containerRef = useRef(null);
+  const [width, setWidth] = useState(0);
+  const {
+    open,
+    x,
+    y,
+    handleClick: hookOnClick,
+    handleMousedown: handleTriggerMousedown,
+    handleClose,
+  } = useAttachedMenu(containerRef);
+
+  function handleTriggerClick() {
+    if (containerRef.current) {
+      const { width: w } = containerRef.current.getBoundingClientRect();
+      setWidth(w);
+      hookOnClick();
+    }
+  }
+
+  function handlePrimaryActionClick(e) {
+    if (onClick) {
+      onClick(e);
+    }
+  }
+
+  const containerClasses = classNames(`${prefix}--combo-button__container`, {
+    [`${prefix}--combo-button__container--open`]: open,
+  });
```

Comment:
we usually take in a custom `className` prop and apply to the outer element, I think we'd want to do that here as well